enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/SlottedPipeBuilder.scala

```diff
@@ -274,28 +274,28 @@ class SlottedPipeBuilder(fallback: PipeBuilder,
   }
 
   private def createProjectionsForResult(columns: Seq[String], pipelineInformation1: PipelineInformation) = {
-    val runtimeColumns: Seq[(String, commandExpressions.Expression)] = columns map {
-      k =>
-        pipelineInformation1(k) match {
-          case LongSlot(offset, false, CTNode, _) =>
-            k -> slottedExpressions.NodeFromSlot(offset)
-          case LongSlot(offset, true, CTNode, _) =>
-            k -> slottedExpressions.NullCheck(offset, slottedExpressions.NodeFromSlot(offset))
-          case LongSlot(offset, false, CTRelationship, _) =>
-            k -> slottedExpressions.RelationshipFromSlot(offset)
-          case LongSlot(offset, true, CTRelationship, _) =>
-            k -> slottedExpressions.NullCheck(offset, slottedExpressions.RelationshipFromSlot(offset))
-
-          case RefSlot(offset, _, _, _) =>
-            k -> slottedExpressions.ReferenceFromSlot(offset)
-
-          case _ =>
-            throw new InternalException(s"Did not find `$k` in the pipeline information")
-        }
-    }
+    val runtimeColumns: Seq[(String, commandExpressions.Expression)] =
+      columns.map(createProjectionForIdentifier(pipelineInformation1))
     runtimeColumns
   }
 
+  private def createProjectionForIdentifier(pipelineInformation1: PipelineInformation)(identifier: String) = {
```

Comment:
How about just `slotInformation`? That `1` itches a little... ;)

Reply:
It was already there, I just extracted a method :) Renaming it something different just because there is an outer instance with the same type seems equally weird IMHO.